ouroboros-consensus-protocol/src/Ouroboros/Consensus/Protocol/Praos/Common.hs

```diff
@@ -108,4 +111,8 @@ data PraosNonces = PraosNonces {
 -- adding to it here under time pressure. See
 -- <https://github.com/input-output-hk/cardano-node/issues/3864>
 class ConsensusProtocol p => PraosProtocolSupportsNode p where
+  type PraosProtocolSupportsNodeCrypto p
```

Comment:
I had hopes that some `p -> c` mapping already existed, but it does not. So I think it's reasonable to include this here, and I also agree it is wise to name is so specifically to this `PraosProtocolSupportsNode` use case, which we haven't given much forethought yet. 👍